src/Image.h

```diff
@@ -157,7 +157,8 @@ class ImageBase {
     /** Get the address of a particular pixel. */
     void *address_of(int x, int y = 0, int z = 0, int w = 0) const {
         uint8_t *ptr = (uint8_t *)origin;
-        size_t offset = x*stride_0 + y*stride_1 + z*stride_2 + w*stride_3;
+        uint64_t offset = (int64_t)x*stride_0 + (int64_t)y*stride_1 +
```

Comment:
This seems like it might be slow on 32-bit platforms. Perhaps size_t instead of uint64_t, and ssize_t instead of int64_t.